addons/html_builder/static/src/core/building_blocks/builder_number_input.js

```diff
@@ -159,7 +166,11 @@ export class BuilderNumberInput extends Component {
     }
 
     get displayValue() {
-        return this.formatRawValue(this.state.value);
+        return this.formatRawValue(this.domState.value);
+    }
+
+    onInput(e) {
+        this.state.showUnit = e.target.value > 0;
```

Comment:
in this case, value == 0 => showUnit == false.


We should check if it s a Number => showUnit